src/util/options.rs

```diff
@@ -426,6 +426,197 @@ impl Options {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum GCTriggerSelector {
+    FixedHeapSize(usize),
+    DynamicHeapSize(usize, usize),
+    Delegated,
+}
+
+impl GCTriggerSelector {
+    const K: u64 = 1024;
+    const M: u64 = 1024 * Self::K;
+    const G: u64 = 1024 * Self::M;
+    const T: u64 = 1024 * Self::G;
+
+    /// Parse a size representation, which could be a number to represents bytes,
+    /// or a number with the suffix K/k/M/m/G/g. Return the byte number if it can be
+    /// parsed properly, otherwise return an error string.
+    fn parse_size(s: &str) -> Result<usize, String> {
+        let s = s.to_lowercase();
+        if s.ends_with(char::is_alphabetic) {
+            let num = s[0..s.len() - 1]
+                .parse::<u64>()
+                .map_err(|e| e.to_string())?;
+            let size = if s.ends_with('k') {
+                num * Self::K
+            } else if s.ends_with('m') {
+                num * Self::M
+            } else if s.ends_with('g') {
+                num * Self::G
+            } else if s.ends_with('t') {
+                num * Self::T
```

Comment:
I still think we should still use `checked_mul` here.  Because it is a user-facing interface, it is possible that the user will supply an unreasonably large number and cause an overflow.

The current Rust Reference says overflow will panic in debug mode, but doesn't specify whether it has undefined behaviour or wrapping behaviour.  But it is better to be safe.

Reply:
Done.